cvat-core/src/server-proxy.ts

```diff
@@ -509,13 +509,31 @@ async function healthCheck(maxRetries, checkPeriod, requestTimeout, progressCall
         timeout: requestTimeout,
     })
         .then((response) => response.data)
-        .catch((errorData) => {
-            if (maxRetries > 0) {
+        .catch((error) => {
+            let isHealthy = true;
+            let data;
+            if ('response' in error && typeof error.response.data === 'object') {
```

Comment:
Something like this should work:

```suggestion
            if (typeof error?.response?.data === 'object') {
```